code/roles/BaseRole.cs

```diff
@@ -46,4 +46,47 @@ public virtual void OnSelect( Player player )
 	public virtual void OnDeselect( Player player ) { }
 
 	public virtual void OnKilled( Player player ) { }
+
+	public static bool operator ==( BaseRole left, BaseRole right )
+	{
+		if ( left is null )
+		{
+			if ( right is null )
+				return true;
+
+			return false;
+		}
+
+		return left.Equals( right );
+	}
+	public static bool operator !=( BaseRole left, BaseRole right ) => !(left == right);
+
+	public static bool operator ==( BaseRole left, string right )
```

Comment:
Can we not ever compare a `BaseRole` to a `string`? If someone does this I don't even know what to say.